src/Core/Framework/Adapter/AdapterException.php

```diff
@@ -166,4 +177,105 @@ public static function invalidArgument(string $message): self
             $message
         );
     }
+
+    public static function currencyFilterError(string $message): self
+    {
+        return new self(
+            Response::HTTP_INTERNAL_SERVER_ERROR,
+            self::CURRENCY_FILTER_ERROR,
+            $message
+        );
+    }
+
+    public static function securityFunctionNotAllowed(string $function): self
+    {
+        return new self(
+            Response::HTTP_FORBIDDEN,
+            self::SECURITY_FUNCTION_NOT_ALLOWED,
+            'Function "{{ function }}" is not allowed',
+            ['function' => $function]
+        );
+    }
+
+    public static function cacheCompressionError(string $message): self
+    {
+        return new self(
+            Response::HTTP_INTERNAL_SERVER_ERROR,
+            self::CACHE_COMPRESSION_ERROR,
+            'Error while processing cache compression. {{ message }}',
+            ['message' => $message],
+        );
+    }
+
+    public static function pcreFunctionError(string $function, string $error): self
+    {
+        return new self(
+            Response::HTTP_INTERNAL_SERVER_ERROR,
+            self::PCRE_FUNCTION_ERROR,
+            'Error while processing Twig {{ function }} function. Error: {{ error }}',
+            ['function' => $function, 'error' => $error]
+        );
+    }
+
+    public static function cacheDirectoryError(string $directory): self
+    {
+        return new self(
+            Response::HTTP_INTERNAL_SERVER_ERROR,
+            self::CACHE_DIRECTORY_ERROR,
+            'Unable to write in the "{{ directory }}" directory',
+            ['directory' => $directory]
+        );
+    }
+
+    public static function currencyFilterMissingContext(): self
+    {
+        return new self(
+            Response::HTTP_INTERNAL_SERVER_ERROR,
+            self::CURRENCY_FILTER_MISSING_CONTEXT,
+            'Error while processing Twig currency filter. No context or locale given.'
+        );
+    }
+
+    public static function currencyFilterMissingIsoCode(): self
+    {
+        return new self(
+            Response::HTTP_INTERNAL_SERVER_ERROR,
+            self::CURRENCY_FILTER_MISSING_ISO_CODE,
+            'Error while processing Twig currency filter. Could not resolve currencyIsoCode.'
+        );
+    }
+
+    /**
+     * @deprecated tag:v6.8.0 - It will return a self instance instead of AdapterFactoryNotFoundException - reason:remove-exception
+     */
+    public static function filesystemFactoryNotFound(string $type): AdapterFactoryNotFoundException|self
+    {
+        if (Feature::isActive('v6.8.0.0')) {
```

Comment:
can we make it the other way around? so with negotiation, in the major preparation we can then just throw away the whole if block. with this, we would need to think about, what needs to stay and what can be removed :see_no_evil: